src/Illuminate/View/Factory.php

```diff
@@ -113,6 +120,7 @@ public function __construct(EngineResolver $engines, ViewFinderInterface $finder
         $this->engines = $engines;
 
         $this->share('__env', $this);
+        $this->share('loop', (object) Arr::last($this->getLoopsStack()));
```

Comment:
The $loop is fixed after construction, how come the $loop is updated in each iteration?